Config.cmake.in

```diff
@@ -1,6 +1,17 @@
 @PACKAGE_INIT@
 
-if(@PROJECT_NAME@_FOUND)
+set(skip_this_file TRUE)
+if(NOT @PROJECT_NAME@_FOUND)
+  set(skip_this_file FALSE)
+endif()
+if(skip_this_file)
+  foreach(component ${@PROJECT_NAME@_FIND_COMPONENTS})
+    if(NOT "@PROJECT_NAME@_${component}_FOUND")
+      set(skip_this_file FALSE)
```

Comment:
I'm afraid this will interact poorly if you are calling the same `find_package` with different components in a project (which I think this is supposed to support) as the targets for the main project will be redefined each time you change the components.

Reply:
I don't know how robust this is but I know for sure that the following works:
```cmake
find_package(foo)
find_package(foo COMPONENTS bar)
```

Reply:
Even if `foo` exports its own targets?

Reply:
Yes, `foo` exports `foo::foo` and component `bar` exports `foo::bar`.

Reply:
After double checking, I think the targets are not redefined because `fooTargets.cmake` defines `foo::foo` only if it is defined and the same holds for `barTargets.cmake`.

Reply:
@jmirabel yes :-)